### PR TITLE
Add support to sync signatures using docker API extension

### LINKED ISSUE #528: As a user, I can sync image signatures from a registry which supports docker/distribution API extension
Synced signatures are not verified. 

If no `sigstore` is configured on a remote and registry supports docker/distribution API extension (aka the header `X-Registry-Supports-Signatures` is set to `"1"`), Pulp should use this API extension to get image signatures from a remote registry.
https://github.com/containers/image/blob/main/docs/signature-protocols.md#openshift-dockerdistribution-api-extension

Sigstore support is implemented as a part of #498

### PR DESCRIPTION
**Pull request by goosemania**
closes #528

### CODE REVIEW

**Review by lubosmj — APPROVED** [review]

File: CHANGES/528.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Added support for syncing signatures using docker API extension.
```
Comment (ipanova) [inline CHANGES/528.feature:1]: now that all the pieces are implemented for the signature sync, can you add/update the docs?

File: pulp_container/app/tasks/sync_stages.py [hunk 2]
```diff
@@ -347,6 +353,29 @@ def _create_manifest_declarative_artifact(self, relative_url, saved_artifact, di
...
+    def _create_signature_declarative_content(
+        self, signature_raw, man_dc, name=None, signature_b64=None
+    ):
+        signature_json = extract_data_from_signature(signature_raw, man_dc.content.digest)
+        if signature_json is None:
+            return
+
+        sig_digest = hashlib.sha256(signature_raw).hexdigest()
+        signature = ManifestSignature(
+            name=name or f"{man_dc.content.digest}@{sig_digest[:32]}",
```
Comment (ipanova) [inline pulp_container/app/tasks/sync_stages.py:365]: :+1:

File: pulp_container/app/tasks/sync_stages.py [hunk 3]
```diff
@@ -459,33 +488,36 @@ async def create_signatures(self, man_dc, signature_source):
...
+                api_extension_signatures = json.loads(signatures_fd.read())
+            for signature in api_extension_signatures.get("signatures", []):
+                if (
+                    signature["schemaVersion"] == SIGNATURE_API_EXTENSION_VERSION
+                    and signature["type"] == SIGNATURE_TYPE.ATOMIC_SHORT
+                ):
+                    signature_base64 = signature["content"]
+                    signature_raw = base64.b64decode(signature_base64)
+                    sig_dc = self._create_signature_declarative_content(
+                        signature_raw, man_dc, signature["name"], signature_base64
```
Comment (ipanova) [inline pulp_container/app/tasks/sync_stages.py:515]: nitpick, while i was reviewing Lubos'es pr -  we should assert that `name` is part of the provided body structure

File: pulp_container/app/tasks/sync_stages.py [hunk 4]
```diff
@@ -459,33 +488,36 @@ async def create_signatures(self, man_dc, signature_source):
...
+                "signatures",
+                man_dc.content.digest,
+            )
+            signatures_downloader = self.remote.get_downloader(url=signatures_url)
+            await signatures_downloader.run()
+            with open(signatures_downloader.path) as signatures_fd:
+                api_extension_signatures = json.loads(signatures_fd.read())
+            for signature in api_extension_signatures.get("signatures", []):
+                if (
+                    signature["schemaVersion"] == SIGNATURE_API_EXTENSION_VERSION
```
Comment (ipanova) [inline pulp_container/app/tasks/sync_stages.py:509]: and maybe signature.get(..) not to stumble on keyerror

File: pulp_container/app/tasks/sync_stages.py (hunk 3 above)
Comment (ipanova) [reply pulp_container/app/tasks/sync_stages.py:515]: or `signature.get('name')

File: docs/workflows/sync.rst [hunk 5]
```diff
@@ -66,6 +66,15 @@ Remote GET Response::
    instead of the whole repository. Note that it is also possible to filter a bunch of tags that
    matches defined criteria by leveraging wildcards.
 
+Some registries contain signed images. Such registries provide signatures in different ways.
+If a registry provides signatures via its SigStore, a URL to it should be specified in
```
Comment (ipanova) [inline docs/workflows/sync.rst:70]: via a separate SigStore

File: docs/workflows/sync.rst [hunk 6]
```diff
@@ -66,6 +66,15 @@ Remote GET Response::
...
+the ``sigstore`` field when creating a Remote.
+
+.. note::
+    Repositories can support the docker API extension for signatures, or have `cosign` signatures.
```
Comment (ipanova) [inline docs/workflows/sync.rst:74]: Some registries provide the docker API extension for `atomic container signature`type , or have `cosign type  signatures`.

**Pull request merged**